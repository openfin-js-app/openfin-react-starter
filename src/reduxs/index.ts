import {combineReducers} from 'redux';
import {IApplicationState} from './application/types';
import {IClientState} from './client/types';
import {IConfigState} from './config/types';

import application from './application/reducer';
import { defaultState as applicationDefaultState } from './application/reducer';
import client from './client/reducer';
import { defaultState as clientDefaultState } from './client/reducer';
import config from './config/reducer';
import { defaultState as configDefaultState } from './config/reducer';

export * from './application/actions';
export * from './application/types';
export * from './client/actions';
export * from './client/types';
export * from './config/actions';
export * from './config/context';
export * from './config/types';

export interface IRootState {
    application:IApplicationState,
    client:IClientState,
    config:IConfigState,
}

export const rootDefaultState = {
    application:applicationDefaultState,
    client:clientDefaultState,
    config:configDefaultState,
}

export default (parentWindowState?:IRootState)=>{
    if (parentWindowState){
        return combineReducers({
            application:application(parentWindowState.application),
            client:client(parentWindowState.client),
            config:config(parentWindowState.config),
        });

    }else{
        return combineReducers({
            application:application(),
            client:client(),
            config:config(),
        });
    }
}
