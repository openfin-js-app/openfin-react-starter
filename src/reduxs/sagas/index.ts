import { all, takeEvery, select } from 'redux-saga/effects';

import clientSaga from './client';

export function* handleLogAllActions(action) {
    const state = yield select();
    console.log(`${action.type}`,action,'state after',state);
}

export function* watchAndLogSaga(){
    yield takeEvery('*',handleLogAllActions)
}

export default function* rootSaga(){

    const sagas = [ clientSaga() ];

    if(process.env.REACT_APP_LOG_ACTION === 'true'){
        sagas.unshift(watchAndLogSaga());
    }

    yield all(sagas);

}