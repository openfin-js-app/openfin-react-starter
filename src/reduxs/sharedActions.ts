import {
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
} from './application/actions';

import {
    CLIENT_SET_VALUE,
} from './client/actions';

const sharedActions:string[]=[
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
    CLIENT_SET_VALUE,
];

export default sharedActions;