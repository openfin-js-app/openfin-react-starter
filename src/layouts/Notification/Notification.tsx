import * as React from 'react';
import { connect } from 'react-redux';
import {Redirect, Route, Switch} from "react-router";
import { makeStyles } from '@material-ui/styles';

import IconButton from '@material-ui/core/IconButton';

import CloseIcon from '@material-ui/icons/Close';

import { Notification } from 'redux-openfin';

import cx from "classnames";

import { notificationStyle as style } from '../../assets/jss/openfin-starter';
import {IRootState, MuiTheme} from '../../reduxs';
import notificationRoutes from "../../routes/notification";
const switchRoutes = (
    <Switch>
        {notificationRoutes.map((prop:any,key)=>{
            if (prop.redirect)
                return <Redirect from={prop.path} to={prop.to} key={key}/>;
            return <Route path={prop.path} component={prop.component} key={key}/>

        })}
    </Switch>
);

interface IProps {
    theme:MuiTheme,
    actions:{
        handleSelfClose:()=>void,
    }
    // for testing
    location?:any,
}

const useStyles = makeStyles(style);

const NotificationLayout:React.FunctionComponent<IProps> = (
    {
        theme,
        actions:{
            handleSelfClose
        }
    }
)=>{

    const classes = useStyles();

    return (
        <div className={
            cx(
                classes.container,
                {
                    [classes.lightBoxShaddow]: theme === MuiTheme.LIGHT
                }
            )
        }>
            <IconButton className={classes.closeBtn}
                        onClick={handleSelfClose}
            >
                <CloseIcon fontSize='small'/>
            </IconButton>
            {switchRoutes}
        </div>
    )
}

export default connect(
    (state:IRootState)=>({
        theme:state.config.application.theme,
    }),
    dispatch => ({
        actions:{
            handleSelfClose:()=>{
                dispatch(Notification.actions.close({}));
            }
        }
    })

    )(
    NotificationLayout
);

