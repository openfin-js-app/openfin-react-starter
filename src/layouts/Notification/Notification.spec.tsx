import * as React from 'react';
import { Provider } from 'react-redux';

import IconButton from '@material-ui/core/IconButton';

import { createShallow, createMount } from '@material-ui/core/test-utils';
import configurestore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import Notification from './Notification';
import {MuiTheme} from "../../reduxs";

const mockStore = configurestore();

const initialState = {
    config:{
        application:{
            theme:MuiTheme.LIGHT,
        }
    }
};

describe('Dashboard layout',()=>{

    let shallow;
    let mount;

    beforeAll(() => {
        mount = createMount();
    });

    afterAll(() => {
        mount.cleanUp();
    });


    beforeEach(() => {
        shallow = createShallow();
    });

    it('render in default state correctly',()=>{

        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/notification']}>
                    <Notification location={{pathname:'/notification'}}/>
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find(Notification)).toHaveLength(1);
        expect(wrapper.find(IconButton)).toHaveLength(1);

        wrapper.find(IconButton).props().onClick();

        expect(store.getActions()).toMatchSnapshot();

    })

})