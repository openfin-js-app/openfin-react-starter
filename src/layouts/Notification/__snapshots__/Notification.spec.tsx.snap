// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification layout render in default state correctly 1`] = `
Array [
  Object {
    "error": false,
    "payload": Object {},
    "type": "REDUX_OPENFIN::NOTIFICATION_CLOSE",
  },
]
`;
